### DISCUSSION TITLE
same-origin policy on API routes not working

### DISCUSSION THREAD

**Discussion by Squirrelcoding** [vercel/next.js#29589] [Help]
### What version of Next.js are you using?

11.1.2

### What version of Node.js are you using?

15.12.0

### What browser are you using?

Chrome (brave)

### What operating system are you using?

Windows

### How are you deploying your application?

Vercel

### Describe the Bug

Say I have a web app with an API route:
```js
export default function handler(req, res) {
  res.status(200).json({ name: 'John Doe' })
}
```
And I deploy it to production, and for some reason I go to https://reqbin.com/ and send an http request to my web app.
```
GET https://my-web-app.vercel.app/api/myAPIRoute
```
And the response I get?
```json
{
 "name": "John Doe"
}
```

### Expected Behavior

To get an error saying that CORS has blocked the request.

### To Reproduce

1. `npx create-next-app`
2. Create an API route
3. Deploy it.
4. make an http request to it whether via curl, code, reqbin.com, etc.

**Comment by ijjk** [comment]
Hi, you can set a more restrictive `access-control-allow-origin` header using the headers config documented here https://nextjs.org/docs/api-reference/next.config.js/headers

**Reply by Squirrelcoding** [reply]
That didn't work, the result is the same

**Reply by dalalRohit** [reply]
Hey, did you get any solution for this? 
I'm also facing the same issue ;(

**Reply by mateosantosdev** [reply]
In my case this code [https://nextjs.org/docs/api-reference/next.config.js/headers](https://nextjs.org/docs/api-reference/next.config.js/headers) works but I was using an old deployment url so I wasn't calling the latest deploy. Be sure you are calling the correct url.
